File: lib/biz/spec-iot/src/main/java/site/addzero/biz/spec/iot/protocol/modbus/ModbusConnectionOptions.java
package site.addzero.biz.spec.iot.protocol.modbus;

/**
 * Connection parameters for a Modbus TCP endpoint.
 */
public final class ModbusConnectionOptions {

    private final String connectionId;
    private final String host;
    private final int port;
    private final String threadId;

    public ModbusConnectionOptions(String connectionId, String host, int port) {
        this(connectionId, host, port, "0");
    }

    public ModbusConnectionOptions(String connectionId, String host, int port, String threadId) {
        this.connectionId = requireText(connectionId, "connectionId");
        this.host = requireText(host, "host");
        this.port = port;
        this.threadId = threadId == null || threadId.trim().isEmpty() ? "0" : threadId.trim();
    }

    public String getConnectionId() {
        return connectionId;
    }

    public String getHost() {
        return host;
    }

    public int getPort() {
        return port;
    }

    public String getThreadId() {
        return threadId;
    }

    String cacheKey() {
        return connectionId + ":" + threadId;
    }

    private static String requireText(String value, String field) {
        if (value == null || value.trim().isEmpty()) {
            throw new IllegalArgumentException(field + " must not be blank");
        }
        return value.trim();
    }
}

File: lib/biz/spec-iot/src/main/kotlin/site/addzero/biz/spec/iot/protocol/modbus/ModbusConnectionOptions.kt
package site.addzero.biz.spec.iot.protocol.modbus

import site.addzero.biz.spec.iot.requireText

/**
 * Connection parameters for a Modbus TCP endpoint.
 */
class ModbusConnectionOptions @JvmOverloads constructor(
    connectionId: String?,
    host: String?,
    val port: Int,
    threadId: String? = "0",
) {

    val connectionId: String = requireText(connectionId, "connectionId")
    val host: String = requireText(host, "host")
    val threadId: String = threadId?.trim()?.takeIf { it.isNotEmpty() } ?: "0"

    internal fun cacheKey(): String {
        return "$connectionId:$threadId"
    }
}
